resources/js/views/settings/roles/Index.vue

```diff
@@ -0,0 +1,232 @@
+            :disabled='isBusy'
+            variant='primary'
+            :to="{ name: 'settings.roles.view', params: { id: data.item.id }, query: { view: '1' } }"
+          >
+            <i class='fas fa-eye'></i>
+          </b-button>
+        </can>
+        <can method='update' :policy='data.item'>
+          <b-button
+            v-b-tooltip.hover
+            :title="$t('settings.roles.edit', { name: data.id })"
+            :disabled='isBusy'
+            variant='dark'
+            :to="{ name: 'settings.roles.view', params: { id: data.item.id } }"
+          >
+            <i class='fas fa-edit'></i>
+          </b-button>
+        </can>
+        <can method='delete' :policy='data.item'>
+          <b-button
+            v-b-tooltip.hover
+            :title="$t('settings.roles.delete.item', { id: data.id })"
+            :disabled='isBusy'
+            variant='danger'
+            @click='showDeleteModal(data.item)'>
+            <i class='fas fa-trash'></i>
+          </b-button>
+        </can>
+      </template>
+    </b-table>
+
+    <b-pagination
+      v-model='currentPage'
+      :total-rows='total'
+      :per-page='perPage'
+      aria-controls='roles-table'
+      @input="$root.$emit('bv::refresh::table', 'roles-table')"
+      align='center'
+      :disabled='isBusy'
+    ></b-pagination>
+
+    <b-modal
+      :busy='isBusy'
+      ok-variant='danger'
+      cancel-variant='dark'
+      :cancel-title="$t('app.false')"
+      @ok='deleteRole'
+      @cancel='clearRoleToDelete'
+      @close='clearRoleToDelete'
+      ref='delete-role-modal'
+      :static='modalStatic'>
+      <template v-slot:modal-title>
+        {{ $t('settings.roles.delete.title') }}
+      </template>
+      <template v-slot:modal-ok>
+        <b-spinner small v-if="isBusy"></b-spinner>  {{ $t('app.true') }}
+      </template>
+      <span v-if="roleToDelete">
+        {{ $t('settings.roles.delete.confirm', { name: $te(`app.roles.${roleToDelete.name}`) ? $t(`app.roles.${roleToDelete.name}`) : roleToDelete.name }) }}
+      </span>
+
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import Base from '../../../api/base';
+import Can from '../../../components/Permissions/Can';
+
+export default {
+  components: { Can },
+
+  props: {
+    modalStatic: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  data () {
+    return {
+      isBusy: false,
+      currentPage: undefined,
+      total: undefined,
+      perPage: undefined,
+      roleToDelete: undefined
+    };
+  },
+
+  computed: {
+    /**
+     * Returns array with table fields and their labels.
+     */
+    tableFields () {
+      return [
+        { key: 'id', label: this.$t('settings.roles.id'), sortable: true },
+        { key: 'name', label: this.$t('settings.roles.name'), sortable: true },
+        { key: 'default', label: this.$t('settings.roles.default'), sortable: true },
+        { key: 'actions', label: this.$t('settings.roles.actions') }
```

Comment:
Hide column if user has not permission to do any actions: 
![image](https://user-images.githubusercontent.com/4281791/94403250-715e4a80-016d-11eb-8103-f55ac64b3e40.png)